### DISCUSSION TITLE
How can I read/write .json file using serverless?

### DISCUSSION THREAD

**Discussion by manhbuithe** [vercel/community#456] [Help] [answered]
Let's say I have a folder named `db`. Inside I have many .json files and use `notarealdb` lib to read, write it.
Everything works well on local but when deploy to Vercel I get empty data.

I tried create `vercel.json` 
```
{
  "functions": {
    "pages/api/movies.js": {
      "includeFiles": "db/**"
    }
  }
}
```

But it didn't work.
I also tried `outputFileTracing: true` but it didn't work too.

Can anyone help me?

<img width="259" alt="image" src="https://user-images.githubusercontent.com/56332920/154728961-f5c92431-2f79-434f-9adc-f86747c59f38.png">
This is the directory.

**Comment by quyctd** [comment]
Same problem here

**Comment by holylander** [comment, marked as answer]
there is no way to write files while running your solution in a vercel sever. You will need to store your solution in another server ( not vercel platform ) or use some other external storage service

**Reply by amyegan** [reply]
An external database provider is definitely a good way to go for this. We have some integrations and example apps to get you started https://vercel.com/docs/concepts/solutions/databases#providers
